### PR TITLE
[REVIEW] Remove ShadowEventInspector

### PR DESCRIPTION
**Pull request by cstamas**
Changes:
- removed nasty ShadowRepositoryEventInspector that put load
  on repo registry, instead made shadows itself "listen" and
  act on their own.
- made shadow repository "cache" it's master instance,
  instead using registry to look it up per every getMasterRepo
  invocation.

These changes above does not disrupt how shadows work at all,
but have a huge benefit that Registry is not used anymore
during event processing shadows are interested in. Registry
is used only once, if master is set using it's ID (that
happens only during boot and/or shadow creation).

### CODE REVIEW

**Comment by cstamas** [comment]
CI
https://bamboo.zion.sonatype.com/browse/NXO-OSSF60-1

**Comment by kellyrob99** [comment]
Is there a Jira to go with this?

File: nexus-core/src/main/java/org/sonatype/nexus/proxy/repository/AbstractShadowRepository.java [hunk 1]
```diff
@@ -128,6 +130,7 @@ public LocalStatus getLocalStatus()
     }
 
     @Override
+    @Subscribe
     public void onRepositoryItemEvent( final RepositoryItemEvent ievt )
```
Comment (jdillon) [inline nexus-core/src/main/java/org/sonatype/nexus/proxy/repository/AbstractShadowRepository.java:134]: where is the corresponding EventBus.register() done for this to work?

Comment (adreghiciu) [inline nexus-core/src/main/java/org/sonatype/nexus/proxy/repository/AbstractShadowRepository.java:134]: This class inherits from AbstractConfigurable that does the registration

Comment (jdillon) [inline nexus-core/src/main/java/org/sonatype/nexus/proxy/repository/AbstractShadowRepository.java:134]: k, thx

**Comment by adreghiciu** [comment]
+1

**Comment by nabcos** [comment]
+1

**Pull request merged**